### PR TITLE
Updating Recurring Donation Change History to Recurring Donation Change Log

### PR DESCRIPTION
**Pull request by lparrott**
- We updated all references of "Recurring Donation Change History" to "Recurring Donation Change Log" to better differentiate between out of the box Field History Tracking.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: force-app/main/default/classes/RD2_ChangeLogSelector.cls [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            String.valueOf(RecurringDonationChangeLog__c.PreviousPaymentMethod__c),
+            String.valueOf(RecurringDonationChangeLog__c.PreviousPlannedInstallments__c),
+            String.valueOf(RecurringDonationChangeLog__c.PreviousRecurringType__c),
+            String.valueOf(RecurringDonationChangeLog__c.PreviousStatus__c),
+            String.valueOf(RecurringDonationChangeLog__c.PreviousStatusReason__c),
+            String.valueOf(RecurringDonationChangeLog__c.PreviousAnnualValue__c),
+            String.valueOf(RecurringDonationChangeLog__c.RecurringDonation__c)
+    };
+
+    public List<RecurringDonationChangeLog__c> getChangeHistories(Id recurringDonationId) {
```
Comment (bdvorachek) [inline force-app/main/default/classes/RD2_ChangeLogSelector.cls:67]: Rename this to getChangeLogEntries?

File: force-app/main/default/classes/RD2_ChangeLogService.cls [hunk 2]
```diff
@@ -31,115 +31,115 @@
...
             }
-            return changeHistoryCreator;
+            return changeLogCreator;
         }
         set;
     }
 
-    private List<RecurringDonationChangeHistory__c> changeHistoryList = 
-        new List<RecurringDonationChangeHistory__c>();
+    private List<RecurringDonationChangeLog__c> changeLogList =
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService.cls:109]: could we rename to `changeLogs`?

File: force-app/main/default/classes/RD2_ChangeLogService.cls [hunk 3]
```diff
@@ -31,115 +31,115 @@
...
+        new List<RecurringDonationChangeLog__c>();
 
     /**
-     * @description Returns true if any of the key fields used to create Change History has been changed.
+     * @description Returns true if any of the key fields used to create Change Logs have been changed.
      * @param updatedRd Changed Recurring Donation
      * @param originalRd Old Recurring Donation
      * @return Boolean
      */
     public void handleHistoryFieldChanged(RD2_RecurringDonation updatedRd, RD2_RecurringDonation originalRd) {
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService.cls:118]: rename to handleLogFieldChanged?

File: force-app/main/default/classes/RD2_ChangeLogService.cls [hunk 4]
```diff
@@ -193,48 +193,48 @@ public inherited sharing class RD2_ChangeHistoryService {
...
-         * @return RecurringDonationChangeHistory__c A new Change History record
+         * @return RecurringDonationChangeLog__c A new Change Log record
          */
-        public RecurringDonationChangeHistory__c initChangeHistoryRecord() {
+        public RecurringDonationChangeLog__c initChangeLogRecord() {
 
             String changeType = calculateChangeType();
 
-            RecurringDonationChangeHistory__c rdChangeHistory = new RecurringDonationChangeHistory__c(
+            RecurringDonationChangeLog__c rdChangeLog = new RecurringDonationChangeLog__c(
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService.cls:237]: rename to `changeLog` without "rd"?

File: force-app/main/default/classes/RD2_ChangeLogService.cls [hunk 5]
```diff
@@ -297,17 +297,17 @@ public inherited sharing class RD2_ChangeHistoryService {
...
 
         /**
-         * @description Inserts the provided Change History records, without allowing partial success
-         * @param changeHistoryList A List of Change History records to insert, regardless of permissions
+         * @description Inserts the provided Change Log records, without allowing partial success
+         * @param changeLogList A List of Change Log records to insert, regardless of permissions
          */
-        public void insertChangeHistoryRecords(List<RecurringDonationChangeHistory__c> changeHistoryList){
-            UTIL_DMLService.insertRecords(changeHistoryList);
+        public void insertChangeLogRecords(List<RecurringDonationChangeLog__c> changeLogList){
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService.cls:309]: rename variable to `changeLogs`?

File: force-app/main/default/classes/RD2_ChangeLogService_TEST.cls [hunk 6]
```diff
@@ -70,8 +70,8 @@ public with sharing class RD2_ChangeHistoryService_TEST {
 
         insert rd;
 
-        List<RecurringDonationChangeHistory__c> rdchList = getRecurringDonationChangeHistory(rd.Id);
-        System.assertEquals(0, rdchList.size(), 'There should be no Change History after RD insert');
+        List<RecurringDonationChangeLog__c> rdclList = getRecurringDonationChangeLog(rd.Id);
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService_TEST.cls:73]: rename to `changeLogs`?

File: force-app/main/default/classes/RD2_ChangeLogService_TEST.cls [hunk 7]
```diff
@@ -90,28 +90,28 @@ public with sharing class RD2_ChangeHistoryService_TEST {
...
+        configureRecurringDonationChangeLogEnabled();
 
         npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact())
             .build();
 
         insert rd;
 
-        List<RecurringDonationChangeHistory__c> rdchList = getRecurringDonationChangeHistory(rd.Id);
-        System.assertEquals(0, rdchList.size(), 'There should be no Change History after RD insert');
+        List<RecurringDonationChangeLog__c> rdclList = getRecurringDonationChangeLog(rd.Id);
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService_TEST.cls:104]: rename to `changeLogs`?

File: force-app/main/default/classes/RD2_ChangeLogService_TEST.cls [hunk 8]
```diff
@@ -120,45 +120,45 @@ public with sharing class RD2_ChangeHistoryService_TEST {
...
-        rdchList = getRecurringDonationChangeHistory(rd.Id);
-        RecurringDonationChangeHistory__c rdch = rdchList[0];
-        System.assertEquals(1, rdchList.size(), 'There should be 1 Change History after the update');
-        System.assertEquals(RD_NEW_AMOUNT, rdch.NewAmount__c, 'New Amount should be ' + RD_NEW_AMOUNT);
-        System.assertEquals(RD_AMOUNT, rdch.PreviousAmount__c, 'Previous Amount should be ' + RD_AMOUNT);
-        System.assertEquals(RD_NEW_AMOUNT * 12, rdch.NewAnnualValue__c, 'New Annual Value should be ' + RD_NEW_AMOUNT * 12);
-        System.assertEquals(RD_AMOUNT * 12, rdch.PreviousAnnualValue__c, 'Previous Annual Value should be ' + RD_AMOUNT * 12);
-        System.assertEquals(RD2_Constants.CHANGE_TYPE_UPGRADE, rdch.ChangeType__c, 
+        rdclList = getRecurringDonationChangeLog(rd.Id);
+        RecurringDonationChangeLog__c rdcl = rdclList[0];
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService_TEST.cls:145]: rename to `changeLog`?

File: force-app/main/default/classes/RD2_ChangeLogService_TEST.cls [hunk 9]
```diff
@@ -208,18 +208,18 @@ public with sharing class RD2_ChangeHistoryService_TEST {
...
+    private static void shouldCreateChangeLogDowngradeWhenPeriodDecreases() {
+        configureRecurringDonationChangeLogEnabled();
 
         npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact())
             .withCalculateNextDonationDate()
             .withInstallmentPeriodWeekly()
             .build();
 
-        RD2_ChangeHistoryService rdchService = new RD2_ChangeHistoryService();
+        RD2_ChangeLogService rdclService = new RD2_ChangeLogService();
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService_TEST.cls:222]: `changeLogService`?

File: force-app/main/default/classes/RD2_ChangeLogService_TEST.cls [hunk 10]
```diff
@@ -180,26 +180,26 @@ public with sharing class RD2_ChangeHistoryService_TEST {
...
         // Check first change
-        List<RecurringDonationChangeHistory__c> rdchList = getRecurringDonationChangeHistory(rd.Id);
-        RecurringDonationChangeHistory__c rdch = rdchList[0];
-        System.assertEquals(1, rdchList.size(), 'There should be 1 Change History after the update');
-        System.assertEquals(RD_NEW_AMOUNT, rdch.NewAmount__c, 'New Amount should be ' + RD_NEW_AMOUNT);
-        System.assertEquals(RD_AMOUNT, rdch.PreviousAmount__c, 'Previous Amount should be ' + RD_AMOUNT);
-        System.assertEquals(RD_NEW_AMOUNT * 12, rdch.NewAnnualValue__c, 'New Annual Value should be ' + RD_NEW_AMOUNT * 12);
-        System.assertEquals(RD_AMOUNT * 12, rdch.PreviousAnnualValue__c, 'Previous Annual Value should be ' + RD_AMOUNT * 12);
-        System.assertEquals(RD2_Constants.CHANGE_TYPE_UPGRADE, rdch.ChangeType__c, 
+        List<RecurringDonationChangeLog__c> rdclList = getRecurringDonationChangeLog(rd.Id);
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_ChangeLogService_TEST.cls:183]: `changeLogs`?

File: force-app/main/default/classes/RD2_OpportunityEvaluationService_TEST.cls [hunk 11]
```diff
@@ -1968,19 +1968,19 @@ private class RD2_OpportunityEvaluationService_TEST {
...
 
-        List<RecurringDonationChangeHistory__c> rdchList = 
-            RD2_ChangeHistoryService_TEST.getRecurringDonationChangeHistory(rd.Id);
-        System.assertEquals(1, rdchList.size(), 'There should be 1 Change History after the update');
-        RecurringDonationChangeHistory__c rdch = rdchList[0];
-        System.assertEquals(RD2_Constants.STATUS_PAUSED, rdch.NewStatus__c, 
+        List<RecurringDonationChangeLog__c> rdclList = 
+            RD2_ChangeLogService_TEST.getRecurringDonationChangeLog(rd.Id);
+        System.assertEquals(1, rdclList.size(), 'There should be 1 Change Log after the update');
+        RecurringDonationChangeLog__c rdcl = rdclList[0];
```
Comment (dfuller114) [inline force-app/main/default/classes/RD2_OpportunityEvaluationService_TEST.cls:1974]: rename to `changeLog`?

**Review by dfuller114 — CHANGES_REQUESTED** [review]
looks good. added several comments to convert abbreviated method and variable names to more clear names where we can.

**Comment by lparrott** [comment]
@bdvorachek @dfuller114 Requested changes have been pushed, thanks for the thorough feedback!

**Review by dfuller114 — APPROVED** [review]
looks good! great work!

**Review by bdvorachek — APPROVED** [review]

**Pull request merged**